### DISCUSSION TITLE
Passwords take kind of long to show

### DISCUSSION THREAD

**Discussion by fourstepper** [zeapo/Android-Password-Store#1264] [General]
Hi,

I have recently switched from a git workflow for pass to using Syncthing for syncing the passwords, thus I had to change to "External repository" in the app.

What I have observed was a major slow down in showing of the passwords available to be searched after opening the app (around 1sec, when before it was basically instantenous)

Is this somewhat a known thing? I don't have that many passwords saved (around 320), and this is slowing my workflow (not badly, but still)

I didn't wanna open an issue for this because I am not sure if this is expected behaviour (perhaps operations above the userspace filesystem take longer?.. I don't really know that much about how android works to be honest)

**Comment by msfjarvis** [comment]
We're currently using the same APIs to access files regardless of where they are stored so that should typically not be a problem, but newer Android versions include extra security checks for files in public storage which can contribute to this slowness when traversing directories with a relatively large number of files.

**Reply by fourstepper** [reply]
I see. Thanks for the reply - that might be it and I am fine with that. In the end, it's still fast enough :) Thank you for your time and answer
